assembly/src/sema/context.rs

```diff
@@ -58,7 +59,11 @@ impl AnalysisContext {
             });
             return Ok(());
         }
-
+        // A string is stored in the constants, w/o trying to evaluate it.
```

Comment:
We can just evaluate it to itself, that will simplify things a bit, and require less special-casing of string constants.